src/VisualStudio/Core/Def/AnalyzerDependency/AnalyzerFileWatcherService.cs

```diff
@@ -1,110 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System;
-using System.Collections.Generic;
-using System.ComponentModel.Composition;
-using System.IO;
-using Microsoft.CodeAnalysis.Host.Mef;
-using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
-using Microsoft.VisualStudio.Shell;
-using Microsoft.VisualStudio.Shell.Interop;
-
-namespace Microsoft.VisualStudio.LanguageServices.Implementation;
-
-[Export(typeof(AnalyzerFileWatcherService))]
-[method: ImportingConstructor]
-[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-internal sealed class AnalyzerFileWatcherService(SVsServiceProvider serviceProvider)
-{
-    private readonly IVsFileChangeEx _fileChangeService = (IVsFileChangeEx)serviceProvider.GetService(typeof(SVsFileChangeEx));
-
-    private readonly Dictionary<string, FileChangeTracker> _fileChangeTrackers = new(StringComparer.OrdinalIgnoreCase);
-
-    /// <summary>
-    /// Holds a list of assembly modified times that we can use to detect a file change prior to the <see cref="FileChangeTracker"/> being in place.
-    /// Once it's in place and subscribed, we'll remove the entry because any further changes will be detected that way.
-    /// </summary>
-    private readonly Dictionary<string, DateTime> _assemblyUpdatedTimesUtc = new(StringComparer.OrdinalIgnoreCase);
-
-    private readonly object _guard = new();
-
-    private static DateTime? GetLastUpdateTimeUtc(string fullPath)
-    {
-        try
-        {
-            var creationTimeUtc = File.GetCreationTimeUtc(fullPath);
-            var writeTimeUtc = File.GetLastWriteTimeUtc(fullPath);
-
-            return writeTimeUtc > creationTimeUtc ? writeTimeUtc : creationTimeUtc;
-        }
-        catch (IOException)
-        {
-            return null;
-        }
-        catch (UnauthorizedAccessException)
-        {
-            return null;
-        }
-    }
-
-    internal void TrackFilePathAndReportErrorIfChanged(string filePath)
```

Comment:
this is the only thing that could cause this type to do anything interesting.  but it has no callers in roslyn.  so this entire type is pointless.

removing.  if someone can find a scenario where this is needed, we can return it.  but we've clearly never noticed that it was doing nothing up till now.